scripts/snip.py

```diff
@@ -158,6 +171,13 @@
 
 with open(os.path.join(snipdir, snipfile), 'w', encoding='utf-8') as f:
```

Comment:
How about only saving the file if it has snippets?
```
# Only generate snip file if it has content.
if len(boilerplates) and len(snippets):
    with open(os.path.join(snipdir, snipfile), 'w', encoding='utf-8') as f:
    ...
```

Reply:
That's what I did at first (but not committed, so not shown here). **However, when they are "sourced" from doc tests, shellcheck (rightly) complains that there is no such file.** To fix some way to check if boilerplate has some bash snippets associated with it. A simpler (but not so good as it introduces coupling with `scripts/gen_snips.sh`) design would be to define "boilerplate snippets" directory from where it can lookup if there is a snippet file (meaning it has some `{{<text>}}` section.

Reply:
As mentioned in comments in `snip.py` we already have some coupling with `scripts/gen_snips.sh`. Maybe I can write that optimization?

Reply:
Done. I applied both optimizations you suggested (and shellcheck is happy)

Reply:
Fixed it. The bug was - consider the case where m boilerplates were found, but 0 snippets. If all m of them don't have snippets then we would have end up with an empty file. It is fixed by filtering boilerplates without snippets early on.